sound/soc/sof/intel/hda.c

```diff
@@ -346,7 +339,17 @@ static int hda_init_caps(struct snd_sof_dev *sdev)
 		dev_err(sdev->dev, "error: no HDMI audio devices found\n");
 		return ret;
 	}
+#endif
+
+	/* Init HDA controller after i915 init */
+	ret = hda_dsp_ctrl_init_chip(sdev, true);
+	if (ret < 0) {
+		dev_err(bus->dev, "error: init chip failed with ret: %d\n",
+			ret);
+		return ret;
+	}
 
```

Comment:
@ranj063 Good catch. 

Just a tiny comments:
My understanding is "snd_hdac_ext_bus_get_ml_capabilities(bus);" should be called after hda_dsp_ctrl_init_chip(sdev, true).